ui/src/base/registry.ts

```
@@ -36,8 +37,23 @@ export class Registry<T> {
     this.key = key;
   }
 
+  set filter(filter: ((key: string) => boolean) | undefined) {
+    this.keyFilter = filter ?? (() => true);
+
+    // Run the filter to knock out anything already registered that does not pass it
+    [...this.registry.keys()]
+      .filter((key) => !this.keyFilter(key))
+      .forEach((key) => this.registry.delete(key));
+  }
```

This also seems a bit odd as API in the following scenario:
 1. Registrants register a variety of items.
 2. Someone sets a filter that allows only subset A through.
 3. Someone sets a filter that allows only subset B through.
 4. Now we end up with only the intersection of subsets A and B, but the filterer at step (3) might reasonably want to get everything in subset B that was ever registered, since the filter for subset A was never what they were interested in (or knew about, or could control, for whatever reason).

That makes me wonder whether it might make sense (and be possible, of course?) to limit what the registry exposes while retaining a reference to the original registrations? Something like 

```
/** retains everything */
registry
/** target for retrieval: contains the subset of items matching current filter */
filteredRegistry
```

---

The purpose is that the "someone" should be an application that embeds Perfetto and is the only authority for such filters. And so wouldn't be inconsistent about it. Nothing in this PR made that clear, of course.

Perhaps this should be a write-once property. Attempts to change the filter after it has been set should throw an error. And then it should be a `setFilter()` method and not a `set filter` prop.

---

Set once behavior would make the expectations explicit 👍